src/aiida/cmdline/commands/cmd_group.py

```diff
@@ -145,41 +145,173 @@ def group_move_nodes(source_group, target_group, force, nodes, all_entries):
+    )
+
+    if groups and filters_provided:
+        echo.echo_critical('Cannot specify both GROUPS and any of the other filters.')
+        return
+
+    if not groups and filters_provided:
+        import datetime
+
+        from aiida.common import timezone
+        from aiida.common.escaping import escape_for_sql_like
+
+        builder = orm.QueryBuilder()
+        filters = {}
 
-    if delete_nodes:
+        # Note: we could have set 'core' as a default value for type_string,
+        # but for the sake of uniform interface, we decided to keep the default value of None.
+        # Otherwise `verdi group delete 123 -T core` would have worked, but we say
+        #  'Cannot specify both GROUPS and any of the other filters'.
+        if type_string is None:
+            type_string = 'core'
 
-        def _dry_run_callback(pks):
-            if not pks or force:
-                return False
-            echo.echo_warning(f'YOU ARE ABOUT TO DELETE {len(pks)} NODES! THIS CANNOT BE UNDONE!')
-            return not click.confirm('Do you want to continue?', abort=True)
+        if '%' in type_string or '_' in type_string:
+            filters['type_string'] = {'like': type_string}
+        else:
+            filters['type_string'] = type_string
+
+        # Creation time
+        if past_days:
+            filters['time'] = {'>': timezone.now() - datetime.timedelta(days=past_days)}
+
+        # Query for specific group labels
+        filters['or'] = []
+        if startswith:
+            filters['or'].append({'label': {'like': f'{escape_for_sql_like(startswith)}%'}})
+        if endswith:
+            filters['or'].append({'label': {'like': f'%{escape_for_sql_like(endswith)}'}})
+        if contains:
+            filters['or'].append({'label': {'like': f'%{escape_for_sql_like(contains)}%'}})
+
+        builder.append(orm.Group, filters=filters, tag='group', project='*')
+
+        # Query groups that belong to specific user
+        if user:
+            user_email = user.email
+        else:
+            # By default: only groups of this user
+            user_email = orm.User.collection.get_default().email
 
-        _, nodes_deleted = delete_group_nodes([group.pk], dry_run=dry_run or _dry_run_callback, **traversal_rules)
-        if not nodes_deleted:
-            # don't delete the group if the nodes were not deleted
+        # Query groups that belong to all users
+        if not all_users:
+            builder.append(orm.User, filters={'email': user_email}, with_group='group')
+
+        # Query groups that contain a particular node
+        if node:
+            builder.append(orm.Node, filters={'id': node.pk}, with_group='group')
+
+        groups = builder.all(flat=True)
+        if not groups:
+            echo.echo_report('No groups found matching the specified criteria.')
             return
 
-    if not dry_run:
+    elif not groups and not filters_provided:
```

Comment:
So, in general it's better to avoid unnecessarily `if` statements, as they are expensive operations. 
I put here below as an `elif` clause, so to avoid hitting it when filters are provided. 
Certainly is not a big effect, but a good practice, I thought :)

Reply:
Ah, OK, I didn't think about this. Wondering if the `if` would actually be a bottleneck here. My idea was rather related to readability, having the cases where one returns immediately in the beginning and at a glance, before delving into all the complicated logic.